File: ecbs-os/backend/src/ECBS.Application/ScreenData/DevicesData.cs
namespace ECBS.Application.ScreenData;

public sealed record DevicesData(
    IReadOnlyList<DeviceDataRow> Devices,
    IReadOnlyList<DeviceKindSummary> Summaries,
    DeviceTelemetrySummary Telemetry,
    string Message,
    string State,
    string UpdatedAt);

public sealed record DeviceDataRow(
    string Id,
    string Name,
    string Kind,
    string SerialNumber,
    string Status,
    string LastSeen,
    string HealthScore,
    string Firmware,
    string Location,
    bool IsMain);

public sealed record DeviceKindSummary(
    string Kind,
    int Total,
    int Online,
    int Offline,
    int Warning);

public sealed record DeviceTelemetrySummary(
    string Kilowatts,
    string KilovoltAmps,
    string KilowattHours,
    string PowerFactor,
    string Timestamp);

File: ecbs-os/frontend/src/lib/devicesData.ts
export type DeviceDataRow = {
  id: string;
  name: string;
  kind: string;
  serialNumber: string;
  status: string;
  lastSeen: string;
  healthScore: string;
  firmware: string;
  location: string;
  isMain: boolean;
};

export type DeviceKindSummary = {
  kind: string;
  total: number;
  online: number;
  offline: number;
  warning: number;
};

export type DeviceTelemetrySummary = {
  kilowatts: string;
  kilovoltAmps: string;
  kilowattHours: string;
  powerFactor: string;
  timestamp: string;
};

export type DevicesData = {
  devices: DeviceDataRow[];
  summaries: DeviceKindSummary[];
  telemetry: DeviceTelemetrySummary;
  message: string;
  state: string;
  updatedAt: string;
};

